src/index.c

```diff
@@ -2633,9 +2633,8 @@ static int write_disk_entry(git_filebuf *file, git_index_entry *entry, const cha
 	if (git_filebuf_reserve(file, &mem, disk_size) < 0)
 		return -1;
 
-	ondisk = (struct entry_short *)mem;
-
-	memset(ondisk, 0x0, disk_size);
+	memset(mem, 0x0, disk_size);
+	memcpy(&ondisk, mem, sizeof(struct entry_short));
```

Comment:
So you first zero out `mem` and then afterwards copy the zeroed out memory into `ondisk`. So simply using `memset(&ondisk, 0, sizeof(ondisk))` would result in more efficient code here, wouldn't it?

Reply:
Yeah, you're right. I was more focusing on the memory-mapped region pointed to by ```mem``` and wanted to make sure it's all cleared out. It was a bit late yesterday when I wrote this ;).

Reply:
On second thought: ```ondisk``` and ```mem``` can have different sizes here: ```sizeof(struct entry_short)``` vs ```disk_size```. So, if I made the change as suggested, the memory region that would get zeroed would be different.